### PR TITLE
PowerShell Console characters garbled for CJK languages

### PR DESCRIPTION
**Pull request by krishnayalavarthi**
This PR applies only to PowerShell 5.1

### CODE REVIEW

File: support/windows-server/system-management-components/PowerShell Console characters garbled for CKJ languages [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+# PowerShell console characters are garbled for Chinese, Japanese and Korean languages on Windows Server 2022
+
+This article provides a resolution to an issue that characters in PowerShell console getting garbled.
+
+_Applies to:_ &nbsp; Windows Server 2022
+_Original KB number:_ &nbsp;
+
+## Symptom
+
+When you launch PowerShell console the characters are getting garbbled. This applies only to Chinese, Japanese and Korean languages SKUs.
```
Comment (sdwheeler) [inline support/windows-server/system-management-components/PowerShell Console characters garbled for CKJ languages:25]: Has a bug been filed for this issue? Also, I assume this is talking about Windows PowerShell 5.1. That should be stated.

Does the problem occur with PowerShell 7?

File: support/windows-server/system-management-components/PowerShell Console characters garbled for CKJ languages [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+
+To fix the issue, there are 3 workarounds listed below. Either of them can be used.
+
+## Workaround 1: Launch the Powershell from cmd.exe
+
+Start cmd.exe and run powershell.exe from within the cmd console. This opens up a PowerShell console without any font issues.
+
+## Workaround 2: Launch the Powershell with this command
+
+Start PowerShell with the command `Powershell -noni` from the search bar. This opens up a PowerShell console without any font issues.
```
Comment (sdwheeler) [inline support/windows-server/system-management-components/PowerShell Console characters garbled for CKJ languages:37]: Starting PowerShell with the `-noninteractive` option is not a real solution. When you do that there are certain features that will not be available. For example, the PSReadLine module will not load so many of the command line editing features will be missing. This is not a recommended solution.

It also tells me that the problem may be in PSReadLine. This should be reported and investigated.

**Review by sdwheeler — CHANGES_REQUESTED** [review]

File: support/windows-server/system-management-components/PowerShell Console characters garbled for CKJ languages (hunk 1 above)
Comment (krishnayalavarthi) [reply support/windows-server/system-management-components/PowerShell Console characters garbled for CKJ languages:25]: There is a bug already filed and fixed. But the release of the fix is in 9.C. Customer is asking for an article until then.

Comment (krishnayalavarthi) [reply support/windows-server/system-management-components/PowerShell Console characters garbled for CKJ languages:25]: This is for PS5.1

File: support/windows-server/system-management-components/PowerShell Console characters garbled for CKJ languages (hunk 2 above)
Comment (krishnayalavarthi) [reply support/windows-server/system-management-components/PowerShell Console characters garbled for CKJ languages:37]: I can remove this option.

File: support/windows-server/system-management-components/PowerShell Console characters garbled for CKJ languages [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+
+To fix the issue, there are 3 workarounds listed below. Either of them can be used.
+
+## Workaround 1: Launch the Powershell from cmd.exe
+
+Start cmd.exe and run powershell.exe from within the cmd console. This opens up a PowerShell console without any font issues.
+
+## Workaround 2: Change the font in the Powershell console
+
+Start PowerShell the normal way. Right click on the Top bar of PowerShell console, select Properties-> Font tab. Goto Font, select `MS Gothic` and click OK. This should resolve the issue.
```
Comment (sdwheeler) [inline support/windows-server/system-management-components/PowerShell Console characters garbled for CKJ languages:37]: Does this only work with `MS Gothic` or would it work with any True Type font?

Does the problem occur with Windows Terminal? If not, that could be another workaround.

**Review by sdwheeler — CHANGES_REQUESTED** [review]

File: support/windows-server/system-management-components/PowerShell Console characters garbled for CKJ languages (hunk 3 above)
Comment (krishnayalavarthi) [reply support/windows-server/system-management-components/PowerShell Console characters garbled for CKJ languages:37]: We tested with `MS Gothic` only and Steve also suggested this. We didn't test with Windows Terminal. I will leave that for now.

Comment (SteveL-MSFT) [reply support/windows-server/system-management-components/PowerShell Console characters garbled for CKJ languages:37]: I believe Windows Terminal has solved this by rendering glyphs from a different font that supports it.  However, I haven't tested this myself.

File: support/windows-server/system-management-components/PowerShell Console characters garbled for CKJ languages (hunk 1 above)
Comment (SteveL-MSFT) [reply support/windows-server/system-management-components/PowerShell Console characters garbled for CKJ languages:25]: I think it's wroth adding the detail that the problem is that the Windows PowerShell .lnk shortcut is hardcoded to Consolas font, by default.  This font doesn't have the glyphs for CJK characters so they aren't rendered correctly.  Changing to MSGothic explicitly will fix this as MSGothic has glyphs for CJK.  The reason cmd.exe works is that the cmd .lnk shortcut doesn't specify a font, so the console chooses the right one at runtime depending on your language.  A similar fix is being made in Windows, but won't be backported.

File: support/windows-server/system-management-components/PowerShell Console characters garbled for CJK languages [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+_Applies to:_ &nbsp; Windows Server 2022
+_Original KB number:_ &nbsp;
+
+## Symptom
+
+When you launch PowerShell console the characters are getting garbbled. This applies only to Chinese, Japanese and Korean languages SKUs.
+
+## Cause
+
+Windows PowerShell .lnk shortcut is hardcoded to Consolas font, by default. This font doesn't have the glyphs for CJK characters so they aren't rendered correctly. Changing to MSGothic explicitly will fix this as MSGothic has glyphs for CJK. The reason cmd.exe works is that the cmd .lnk shortcut doesn't specify a font, so the console chooses the right one at runtime depending on your language. A similar fix is being made in Windows, but won't be backported.
```
Comment (SteveL-MSFT) [inline support/windows-server/system-management-components/PowerShell Console characters garbled for CJK languages:29]: Should specify which version of Windows will get the fix?

Comment (krishnayalavarthi) [reply support/windows-server/system-management-components/PowerShell Console characters garbled for CJK languages:29]: I will add these details.

**Comment by PRMerger13** [comment]
@Deland-Han : Thanks for your contribution! The author(s) have been notified to review your proposed change.

**Comment by PRMerger7** [comment]
@Deland-Han : Thanks for your contribution! The author(s) have been notified to review your proposed change.

**Pull request merged**